File: Neeraj-Sujan/1232-check-if-it-is-a-straight-line.py
from typing import *

class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        if(len(coordinates) == 2):
            return True
        
        coordinates = sorted(coordinates, key=lambda x: x[0])
        
        
        numerator = coordinates[1][1] - coordinates[0][1]
        denominator = coordinates[1][0] - coordinates[0][0]
        
        for i in range(0, len(coordinates)-1):
            p, q = coordinates[i][0], coordinates[i][1]
            r, s = coordinates[i+1][0], coordinates[i+1][1]
            
            if (numerator * (r-p) != (denominator * (s-q))):
                return False
           
        return True


if __name__ == "__main__":
    coordinates = [[1,2],[2,3],[3,4],[4,5],[5,6],[6,7]]
    soln = Solution()

    res = soln.checkStraightLine(coordinates=coordinates)
    print(res)

File: Rishant-Jain/1232. Check If It Is a Straight Line.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& A) {
        int n = A.size();
        if (n <=2 ) return 1;
        
        for (int i = 0 ; i < n - 2 ; i++) {
            int x1 = A[i][0], y1 = A[i][1];
            int x2 = A[i+1][0], y2 = A[i+1][1];
            int x3 = A[i+2][0], y3 = A[i+2][1];
            
            if ((y2 - y1)*(x3 - x2) != (y3 - y2)*(x2 - x1)) return 0;
        }
        return 1;
    }
};